spec/services/taxonomy/taxonomy_query_spec.rb

```diff
@@ -0,0 +1,173 @@
+        expect(TaxonomyQuery.new.tagged_to_taxons(%w[taxon_id_1 taxon_id_2 taxon_id_3], slice_size: 2))
+            .to eq(%w[content_id_1 content_id_2 content_id_3])
+      end
+      it 'removes duplicates' do
+        allow(Services).to receive(:rummager_enum)
+                               .and_return [{ 'content_id' => 'content_id_1' }, { 'content_id' => 'content_id_1' }].to_enum
+        expect(TaxonomyQuery.new.tagged_to_taxons(['id'])).to eq(['content_id_1'])
+      end
+    end
+
+    describe '#taxonomy' do
+      context 'there are no root taxons' do
+        before :each do
+          allow(Services.content_store).to receive(:content_item).with('/').and_return no_taxons
+        end
+        it 'returns an empty taxonomy' do
+          expect(TaxonomyQuery.new.taxonomy(:content_id, :base_path)).to be_empty
+        end
+      end
+
+      context 'there are root taxons and one level of children' do
+        before :each do
+          allow(Services.content_store).to receive(:content_item).with('/').and_return root_taxons
+          allow(Services.content_store).to receive(:content_item).with('/taxons/root_taxon_a')
+                                             .and_return single_level_child_taxons('root_taxon_a', 'child_a_1', 'child_a_2')
+          allow(Services.content_store).to receive(:content_item).with('/taxons/root_taxon_b')
+                                             .and_return single_level_child_taxons('root_taxon_b', 'child_b_1', 'child_b_2')
+        end
+
+        it 'returns root taxons in the first array' do
+          expect(TaxonomyQuery.new.taxonomy(:content_id, :base_path).first)
+            .to match_array [{ 'content_id' => 'root_taxon_a', 'base_path' => '/taxons/root_taxon_a' },
+                             { 'content_id' => 'root_taxon_b', 'base_path' => '/taxons/root_taxon_b' }]
+        end
+
+        it 'returns the first level of child taxons after the root taxons' do
+          expect(TaxonomyQuery.new.taxonomy(:content_id, :base_path).second)
+            .to match_array [{ 'content_id' => 'child_a_1', 'base_path' => '/taxons/child_a_1' },
+                             { 'content_id' => 'child_a_2', 'base_path' => '/taxons/child_a_2' },
+                             { 'content_id' => 'child_b_1', 'base_path' => '/taxons/child_b_1' },
+                             { 'content_id' => 'child_b_2', 'base_path' => '/taxons/child_b_2' }]
+        end
+      end
+
+      context 'there are root taxons and two levels of children' do
+        before :each do
+          allow(Services.content_store).to receive(:content_item).with('/').and_return root_taxon
+          allow(Services.content_store).to receive(:content_item).with('/taxons/root_taxon')
+                                             .and_return multi_level_child_taxons
+        end
+        it 'returns three levels' do
+          expect(TaxonomyQuery.new.taxonomy(:content_id, :base_path).size).to eq(3)
```

Comment:
Reading this without any knowledge of the internals of `TaxonomyQuery#taxonomy`, I'd expect this method to be called `depth`. I find "size of taxonomy" to be ambiguous, could it be depth, width, number of taxons?

Reply:
TaxonomyQuery.new.taxonomy returns an Array of numbers - to have a 'depth' method I'd have to have it return an instance of a custom class, which seems a bit heavyweigth

Reply:
It might be something we want to have, though perhaps I'm the only one that finds the class interface quite difficult to understand what going to be returned. 

Using one of the `taxonomy_query_spec.rb` examples, we seem to be returning arrays of hashes and then relying on Ruby's array methods to return the types of queries we're after.

```rb
taxonomy = TaxonomyQuery.new.taxonomy
=> [
  [
    {"content_id"=>"root_taxon_a", "base_path"=>"/taxons/root_taxon_a"},
    {"content_id"=>"root_taxon_b", "base_path"=>"/taxons/root_taxon_b"}
  ],
  [
    {"content_id"=>"child_a_1", "base_path"=>"/taxons/child_a_1"},
    {"content_id"=>"child_a_2", "base_path"=>"/taxons/child_a_2"},
    {"content_id"=>"child_b_1", "base_path"=>"/taxons/child_b_1"},
    {"content_id"=>"child_b_2", "base_path"=>"/taxons/child_b_2"}
  ]
]

taxonomy.size
=> 2

taxonomy.first
=> [
  {"content_id"=>"root_taxon_a", "base_path"=>"/taxons/root_taxon_a"},
  {"content_id"=>"root_taxon_b", "base_path"=>"/taxons/root_taxon_b"}
]

```

Perhaps renaming this class to something like `TaxonomyDepthQuery` that has public methods `depth`, `taxons_at_level(n)` might clarify how to use the query?